post/post_closing_message.js

```diff
@@ -0,0 +1,117 @@
+const asyncAuto = require('async/auto');
+const {getNode} = require('ln-service');
+const {getChannels} = require('ln-service');
+const {returnResult} = require('asyncjs-util');
+
+const channelPoint = (txid, tx_vout) => `${txid}:${tx_vout}`;
+const escape = text => text.replace(/[_[\]()~`>#+\-=|{}.!\\]/g, '\\\$&');
+const markup = {parse_mode: 'MarkdownV2'};
+const textJoiner = '\n';
+const tokensAsBigTok = tokens => (tokens / 1e8).toFixed(8);
+
+/** Post a channel closing message for Telegram
+
+  {
+    from: <Node From String>
+    id: <Connected Telegram User Id String>
+    lnd: <Authenticated LND API Object>
+    transaction_id: <Funding transaction Id String>
+    transaction_vout: <Funding transaction output Index>
+    send: <Send Message to Telegram User Function>
+  }
+
+  @returns via cbk or Promise
+  {
+    text: <Channel Closing Message Text String>
+  }
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.from) {
+          return cbk([400, 'ExpectedFromNodeToPostClosingMessage']);
+        }
+
+        if (!args.id) {
+          return cbk([400, 'ExpectedConnectedUserIdToPostClosingMessage'])
+        }
+
+        if (!args.lnd) {
+          return cbk([400, 'ExpectedAuthenticatedLndToPostClosingMessage']);
+        }
+
+        if (!args.send) {
+          return cbk([400, 'ExpectedSendFunctionToPostClosingMessage']);
+        }
+
+        if (!args.transaction_id) {
+          return cbk([400, 'ExpectedTransactionIdToPostClosingMessage']);
+        }
+
+        if (args.transaction_vout === undefined) {
+          return cbk([400, 'ExpectedTransactionVoutToPostClosingMessage']);
+        }
+
+        return cbk();
+      },
+
+      //Get closing channel details
+      getClosingChannel: ['validate', ({}, cbk) => {
+          return getChannels({lnd: args.lnd}, (err, res) => {
+            if (!!err) {
+              return cbk([400, 'FailedToGetChannelsToPostClosingMessage', err]);
+            }
+
+            const closing = res.channels.find(n => {
+              return n.transaction_id === args.transaction_id;
+            });
+
+            if (!closing) {
+              return cbk([400, 'FailedToFindChannelToPostClosingMessage']);
+            }
+  
+            return cbk(null, closing);
+          });
+      }],
+
+      //Get Closing peer
+      getClosingPeer: ['getClosingChannel', ({getClosingChannel}, cbk) => {
+        return getNode({lnd: args.lnd, public_key: getClosingChannel.partner_public_key}, (err, res) => {
```

Comment:
this should use `getNodeAlias` otherwise it will fail on error and have to get all the channels